File: app/src/main/java/sk/styk/martin/apkanalyzer/activity/permission/PermissionDetailFragment.java
package sk.styk.martin.apkanalyzer.activity.permission;

import android.os.Bundle;
import android.support.design.widget.CollapsingToolbarLayout;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import sk.styk.martin.apkanalyzer.R;
import sk.styk.martin.apkanalyzer.adapter.pager.PermissionsPagerAdapter;
import sk.styk.martin.apkanalyzer.model.detail.PermissionData;
import sk.styk.martin.apkanalyzer.model.permissions.LocalPermissionData;
import sk.styk.martin.apkanalyzer.model.permissions.PermissionStatus;

public class PermissionDetailFragment extends Fragment {

    public static final String TAG = PermissionDetailFragment.class.getSimpleName();

    public static final String ARG_PERMISSIONS_DATA = "permission_args";
    public static final String ARG_CHILD = "permission_args_to_my_sweetest_child";

    private LocalPermissionData permissionData;

    private CollapsingToolbarLayout appBarLayout;
    private TextView toolbarDescription;

    private PermissionsPagerAdapter adapter;
    private ViewPager viewPager;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        adapter = new PermissionsPagerAdapter(getActivity(), getFragmentManager());
        permissionData = getArguments().getParcelable(ARG_PERMISSIONS_DATA);

        adapter.dataChange(permissionData);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_permission_detail, container, false);

        appBarLayout = getActivity().findViewById(R.id.toolbar_layout);
        toolbarDescription = getActivity().findViewById(R.id.toolbar_layout_description);

        appBarLayout.setTitle(permissionData.getPermissionData().getName());
        toolbarDescription.setText(permissionData.getPermissionData().getSimpleName());

        viewPager = rootView.findViewById(R.id.pager);
        viewPager.setAdapter(adapter);

        TabLayout tabLayout = rootView.findViewById(R.id.tabs);
        tabLayout.setupWithViewPager(viewPager);

        return rootView;
    }
}

File: app/src/main/java/sk/styk/martin/apkanalyzer/ui/permission/detail/pager/PermissionDetailFragment.kt
package sk.styk.martin.apkanalyzer.ui.permission.detail.pager

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import dagger.hilt.android.AndroidEntryPoint
import sk.styk.martin.apkanalyzer.databinding.FragmentPermissionDetailBinding
import sk.styk.martin.apkanalyzer.core.apppermissions.model.LocalPermissionData
import sk.styk.martin.apkanalyzer.util.components.toDialog
import sk.styk.martin.apkanalyzer.util.materialContainerTransform
import sk.styk.martin.apkanalyzer.util.provideViewModel
import javax.inject.Inject

@AndroidEntryPoint
class PermissionDetailFragment : Fragment() {

    @Inject
    lateinit var viewModelFactory: PermissionDetailFragmentViewModel.Factory

    private lateinit var binding: FragmentPermissionDetailBinding

    private lateinit var viewModel: PermissionDetailFragmentViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sharedElementEnterTransition = requireContext().materialContainerTransform()
        viewModel = provideViewModel { viewModelFactory.create(requireNotNull(requireArguments().getParcelable(ARG_PERMISSIONS_DATA))) }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        binding = FragmentPermissionDetailBinding.inflate(inflater, container, false)
        binding.pager.adapter = PermissionDetailPagerAdapter(requireArguments(), requireContext().applicationContext, childFragmentManager)
        binding.tabs.setupWithViewPager(binding.pager)
        binding.lifecycleOwner = viewLifecycleOwner
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.viewModel = viewModel

        with(viewModel) {
            showDialog.observe(viewLifecycleOwner) {
                it.toDialog().show(parentFragmentManager, "PermissionDescription")
            }
            close.observe(viewLifecycleOwner) { requireActivity().onBackPressed() }
        }
    }

    companion object {

        const val ARG_PERMISSIONS_DATA = "permission_args"

        fun create(permissionData: LocalPermissionData) = PermissionDetailFragment().apply {
            arguments = Bundle().apply {
                putParcelable(ARG_PERMISSIONS_DATA, permissionData)
            }
        }
    }
}
